tasmota/tasmota_xnrg_energy/xnrg_25_mk_sky_blu.ino

```diff
@@ -0,0 +1,783 @@
+  uint8_t timeout;                      // active after a request, waiting for reponse
+#ifdef _WITH_ON_OFF_SUPPORT
+  bool actual_state;                    // true = active, false = stop
+  bool target_state;                    // true = active, false = stop
+#endif
+} mksb_data_t;
+
+const char * mksb_register_names[] = {
+  "0 unknown",                 //     dummy ? 
+  "1 MPPT bulk [V]",           // D02 MPPT Voltage limit BULK, >= stops charging
+  "2 MPPT float [V]",          // D01 MPPT voltage limit FLOAT, <= restarts charging, SLA battery only
+  "3 Ontime load [h]",         // D00 Time duration the load gets connected, default: 24h 
+  "4 MPPT current max.[A]",    //  -  MPPT current limit, default 30A, 40A, 50A or 60A
+                               // CAUTION: do not set above the limit of your specific hardware
+  "5 Batt.UVP cutload [V]",    // D03 Battery UnderVoltageProtection, <= cuts load
+  "6 Batt.UVP connload [V]",   //  ~  Battery UnderVoltageProtection, >=limit reconnects load, typ D03 + 0.2V
+  "7 Comm. address (?)",       //  -  Communication Address (ro)
+  "8 Batt.Type [0=SLA, 1=Li]", // D04 Battery Type: value 0=SLA, 1=LiPo (ro)
+  "9 Batt.System [n*12V]"      //  -  Battery System: 1...4 * 12V (Read-Only, set at Batt.connection)
+};
+
+//  const char MKSB_HTTP_SNS_sS_m_De[]       PROGMEM = "{s}%s " "{m}" " %d{e}";
+//  const char MKSB_HTTP_SNS_str_m_str[]       PROGMEM = "{s}%s " "{m}" " %s{e}";
+//  const char MKSB_HTTP_SNS_str_int_str[]        PROGMEM = "{s}%s " "%d{m}" "%s{e}";
+  const char MKSB_HTTP_SNS_str_m_int[]          PROGMEM = "{s}%s " "{m}" "%d{e}";
+
+/*
+const char * mksb_status_bits[] = {
+                                    // mode
+  "",                               // b0 
+  "",                               // b1
+  D_CHARGE,                         // b2 MPPT Mode Active
+  "",                               // b3
+
+  "",                               // b4
+  "",                               // b5
+  "",                               // b6
+  "",                               // b7
+
+                                            // error
+  D_BATTERY " " D_VOLTAGE " " D_STR_LOW,    // b0 Battery undervoltage (< register 5 / 6)
+  D_BATTERY " " D_VOLTAGE " " D_STR_HIGH,   // b1 Battery overvoltage (> register 1)
+  "",                                       // b2
+  "",                                       // b3
+
+  "",                                       // b4
+  "",                                       // b5
+  "",                                       // b6
+  ""                                        // b7
+};
+*/
+
+#ifdef _WITH_PHASE_NAME_SUPPORT
+const char * mksb_channels[] = {D_DEVICE_INPUT , D_BATTERY};
+#endif
+
+#ifdef _WITH_SERIAL_DEBUGGING
+const char * mksb_rx_counter_names[] = {
+  "Good",
+  "Timeout",
+  "TooShort",
+  "Overflow",
+  "CRC",
+  "Command",
+  "Length"
+};
+#endif
+
+// module data
+mksb_data_t me = { .Serial = nullptr, .pRxBuffer = nullptr }; // singleton
```

Comment:
Prefer a pointer, so the structure is allocated only if the driver is really used. Otherwise it consumes precious RAM even when the driver is not used

Also, don't call it `me` which far too generic, prefer some name like `mksb` or similar